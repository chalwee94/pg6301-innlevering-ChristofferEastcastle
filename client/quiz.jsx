import {
  BrowserRouter,
  Link,
  Route,
  Routes,
  useNavigate,
} from "react-router-dom";
import { useEffect, useState } from "react";
import axios from "axios";

function FrontPage() {
  return (
    <div>
      <h1>QuizApplication</h1>
      <ul>
        <li>
          <Link to={"/question"}>Answer question</Link>
        </li>
      </ul>
    </div>
  );
}

function QuestionPage() {
  return (
    <div>
      <h1>Question</h1>
      <p>/*Click on the one you think is right*/</p>
      <Question />
    </div>
  );
}

function Question({ correctAnswer, answered }) {
  const [question, setQuestion] = useState();

  useEffect(async () => {
    setQuestion(await getQuestion());
  }, []);

  async function getQuestion() {
    return await axios.get("http://localhost:3000/api/question");
  }

  let navigate = useNavigate();

  function handleAnswer(answer) {
    answered((a) => a + 1);
    if (isCorrectAnswer(question, answer)) {
      correctAnswer((c) => c + 1);
      navigate("/answer/correct");
    } else {
      navigate("/answer/wrong");
    }
  }

  return (
    <div>
      {/*
      <h4>{question.question}</h4>
      {Object.keys(question.answers).map((k) => (
        <p onClick={() => handleAnswer(k)} key={k}>
          {question.answers[k]}
        </p>
      ))}
      */}
    </div>
  );
}

export function QuizApplication() {
  return (
    <BrowserRouter>
      <Routes>
        <Route path={"/"} element={<FrontPage />} />
        <Route path={"/question"} element={<QuestionPage />} />
      </Routes>
    </BrowserRouter>
  );
}
