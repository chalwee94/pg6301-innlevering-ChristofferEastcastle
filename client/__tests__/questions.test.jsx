import * as React from "react";
import ReactDOM from "react-dom";

import {QuizApplication} from "../quiz.jsx";


describe("quiz application", () => {


  it("shows front page", () => {
    const element = document.createElement("div");
    ReactDOM.render(<QuizApplication/>, element)

    expect(element.querySelector("h1").innerHTML)
      .toEqual("QuizApplication")
    expect(element.innerHTML).toMatchSnapshot()
  })
})