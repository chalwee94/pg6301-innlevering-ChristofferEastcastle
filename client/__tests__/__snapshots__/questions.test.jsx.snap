// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`quiz application shows front page 1`] = `"<div><h1>QuizApplication</h1><h3>Correct answers: 0</h3><h3>Total answered: 0</h3><ul><li><a href=\\"/question\\">Answer question</a></li></ul></div>"`;

exports[`quiz application shows mock question 1`] = `"<div><h4>Which of the following are core Kubernetes objects?</h4><p>Pods, Services, Volumes</p><p>Pods, Docker, Volumes</p><p>Pods, Services, Droplets</p><p>None of the mentioned</p><p></p><p></p></div>"`;
